android/app/src/main/java/com/epfl/dedis/hbt/utility/json/JsonType.kt

```diff
@@ -1,6 +1,24 @@
 package com.epfl.dedis.hbt.utility.json
 
-enum class JsonType(val schemaPath: String) {
-    COMPLETE_TRANSACTION("protocol/complete_transaction.json"),
-    PENDING_TRANSACTION("protocol/pending_transaction.json")
+import com.epfl.dedis.hbt.data.model.CompleteTransaction
+import com.epfl.dedis.hbt.data.model.PendingTransaction
+import kotlin.reflect.KClass
+
+sealed class JsonType<T : Any>(val schemaPath: String, val type: KClass<T>) {
+    object CompleteTransactionType :
```

Comment:
I would name it CompleteTransaction, as when you use it, you would write JsonType.... we already know it's a type.